tests/unit/test_paragraphHelper.py

```diff
@@ -0,0 +1,207 @@
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_multiLineBreakParagraph_forward(self):
+		"""
+			Along with double line break paragraphs (block style),
+			the algorithm supports any number of blank lines between block paragraphs.
+			It also supports white space in either the blank lines,
+			or preceeding the block paragraph itself.
+		"""
+		text = """
+
+
+Paragraph 1.
+
+
+\tParagraph 2.
+
+
+
+  \t\t\t
+
+
+     Paragraph 3.
+
+
+
+		"""
+		obj = BasicTextProvider(text=text)
+		ti = obj.makeTextInfo(textInfos.POSITION_FIRST)
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "\tParagraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # third paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "     Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(True, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_multiLineBreakParagraph_backward(self):
+		"""
+			Test moving back when paragraphs are separated by multiple blank lines / whitespace
+		"""
+		text = """
+
+
+Paragraph 1.
+
+
+\tParagraph 2.
+
+
+
+  \t\t\t
+
+
+     Paragraph 3.
+
+
+
+		"""
+		obj = BasicTextProvider(text=text)
+		ti = obj.makeTextInfo(textInfos.POSITION_LAST)
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # third paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "     Paragraph 3.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # second paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "\tParagraph 2.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # first paragraph
+		self.assertTrue(moved)
+		ti.expand(textInfos.UNIT_LINE)
+		self.assertEqual(ti.text, "Paragraph 1.\n")
+		moved = paragraphHelper._moveTextInfoToBlockParagraph(False, ti)  # no more paragraphs
+		self.assertFalse(moved)
+
+	def test_DoubleLineBreakWithSingleLineBreakContent(self):
```

Comment:
I think testing the reverse `test_SingleLineBreakMovement_WithDoubleLineBreakContent` might also be good

Reply:
@seanbudd I can test this, but it is just going to stop on blank lines as well as populated lines, so I'm not sure what it proves.

Reply:
If a user is using single line break movement, and navigates multi-line content, this test clearly demonstrates the UX.
This shows the UX improvement of switching between modes.